hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -495,17 +499,27 @@ protected boolean containsKeyInternal(Data key) {
         }
     }
 
+
+
     public void waitUntilLoaded() {
         try {
             int mapNamePartition = partitionService.getPartitionId(name);
+            if (Thread.currentThread() instanceof PartitionOperationThread) {
```

Comment:
I wasn't sure about the NearCacheProxy but thx for the explanation. I removed the check.